src/Files.App/UserControls/Widgets/QuickAccessWidget.xaml.cs

```diff
@@ -144,64 +166,85 @@ public bool ShowMultiPaneControls
 			}
 		}
 
-		public string WidgetName => nameof(FolderWidget);
+		public string WidgetName => nameof(QuickAccessWidget);
 
-		public string AutomationProperties => "FolderWidgetAutomationProperties/Name".GetLocalizedResource();
+		public string AutomationProperties => "QuickAccess".GetLocalizedResource();
 
-		public string WidgetHeader => "Folders".GetLocalizedResource();
+		public string WidgetHeader => "QuickAccess".GetLocalizedResource();
 
-		private async void FolderWidget_Loaded(object sender, RoutedEventArgs e)
+		private async void ModifyItem(object? sender, ModifyQuickAccessEventArgs? e)
 		{
-			Loaded -= FolderWidget_Loaded;
+			if (e is null)
+				return;
 
-			ItemsAdded.Add(new FolderCardItem("Desktop".GetLocalizedResource())
-			{
-				Path = UserDataPaths.GetDefault().Desktop,
-				SelectCommand = LibraryCardCommand
-			});
-			ItemsAdded.Add(new FolderCardItem("Documents".GetLocalizedResource())
+			if (e.Add)
 			{
-				Path = UserDataPaths.GetDefault().Documents,
-				SelectCommand = LibraryCardCommand
-			});
-			ItemsAdded.Add(new FolderCardItem("Downloads".GetLocalizedResource())
-			{
-				Path = UserDataPaths.GetDefault().Downloads,
-				SelectCommand = LibraryCardCommand
-			});
-			ItemsAdded.Add(new FolderCardItem("Music".GetLocalizedResource())
-			{
-				Path = UserDataPaths.GetDefault().Music,
-				SelectCommand = LibraryCardCommand
-			});
-			ItemsAdded.Add(new FolderCardItem("Pictures".GetLocalizedResource())
-			{
-				Path = UserDataPaths.GetDefault().Pictures,
-				SelectCommand = LibraryCardCommand
-			});
-			ItemsAdded.Add(new FolderCardItem("Videos".GetLocalizedResource())
+				var locationItems = new List<LocationItem>();
+				foreach (var item in e.Paths)
+					locationItems.Add(await App.QuickAccessManager.Model.CreateLocationItemFromPathAsync(item));
+
+				foreach (var item in locationItems)
+					ItemsAdded.Insert(e.Pin ? ItemsAdded.Count - 4 : ItemsAdded.Count, new FolderCardItem(Path.GetFileName(item.Text), e.Pin) // Add just after the Recent Folders
```

Comment:
Check if there are at least 4 items before removing them?

Reply:
Yes, this is because Explorer shows the last 4 items in the sidebar which are actually recent folders

Reply:
I meant show we verify there are at least 4 items before trying to remove them? I think this caused a crash in the sidebar.